Fix usage writing when using custom version flag

hhromic opened the pull request:
Hi @alexflint , this is a small follow-up to PR #223 :)

I realized that I forgot to update the write usage function, and it was still printing the output of `Version()` even if a custom `--version` flag spec is present in the arguments. This is now also fixed and I added more unit tests for help/usage writing.

I also took the opportunity to improve a bit the way version flag specs are scanned for both usage and help writing.

I belive that this PR now fully addresses issue #171 because, when using a custom `--version` flag, the builtin version string (if any) is not printed anymore in usage and errors.

Fixes #171

hhromic commented:
Rebased on top of the latest `master` branch.
@alexflint any chance of taking a look a this PR? :)
Thanks!

alexflint commented:
@hhromic This looks good but I need to look closer because there are quite a few changes (even ignoring the tests -- thanks for adding those)

hhromic commented:
Hi @alexflint !
I rebased this PR on top of the current master branch in case you are still interested on this fix.
I also refactored the unit tests to be closer to the coding style you had already.

alexflint commented:
Wonderful, thanks for doing this @hhromic
